# Video Game Sales

`Matheus Raz (mrol@cin.ufpe.br)`

`João Paulo Lins (jplo@cin.ufpe.br)`

## Aprendizagem (etapa 2)

In [1]:
from IPython.display import display

import numpy as np
import pandas as pd
from sklearn.neural_network import MLPClassifier
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import accuracy_score

In [2]:
df = pd.read_csv('vgsalesPP.csv')
df

Unnamed: 0                                               Name Platform  \
0               0                                         Wii Sports      Wii   
1               1                                  Super Mario Bros.      NES   
2               2                                     Mario Kart Wii      Wii   
3               3                                  Wii Sports Resort      Wii   
4               4                           Pokemon Red/Pokemon Blue       GB   
5               5                                             Tetris       GB   
6               6                              New Super Mario Bros.       DS   
7               7                                           Wii Play      Wii   
8               8                          New Super Mario Bros. Wii      Wii   
9               9                                          Duck Hunt      NES   
10             10                                         Nintendogs       DS   
11             11                                      Mario Kart DS       DS   
12             12                        Pokemon Gold/Pokemon Silver       GB   
13             13                                            Wii Fit      Wii   
14             14                                 Kinect Adventures!     X360   
15             15                                       Wii Fit Plus      Wii   
16             16                                 Grand Theft Auto V      PS3   
17             17                      Grand Theft Auto: San Andreas      PS2   
18             18                                  Super Mario World     SNES   
19             19       Brain Age: Train Your Brain in Minutes a Day       DS   
20             20                      Pokemon Diamond/Pokemon Pearl       DS   
21             21                                   Super Mario Land       GB   
22             22                                Super Mario Bros. 3      NES   
23             23                                 Grand Theft Auto V     X360   
24             24                        Grand Theft Auto: Vice City      PS2   
25             25                      Pokemon Ruby/Pokemon Sapphire      GBA   
26             26        Brain Age 2: More Training in Minutes a Day       DS   
27             27                        Pokemon Black/Pokemon White       DS   
28             28                             Gran Turismo 3: A-Spec      PS2   
29             29                     Call of Duty: Modern Warfare 3     X360   
...           ...                                                ...      ...   
16382       16382              Help Wanted: 50 Wacky Jobs (jp sales)      Wii   
16383       16383  Yattaman Wii: BikkuriDokkiri Machine de Mou Ra...      Wii   
16384       16384                              Neo Angelique Special      PSP   
16385       16385                      Outdoors Unleashed: Africa 3D      3DS   
16386       16386                                          Real Rode      PS2   
16387       16387                                     Pony Friends 2       PC   
16388       16388                       Dynasty Warriors: Eiketsuden      PS3   
16389       16389                  Metal Gear Solid V: Ground Zeroes       PC   
16390       16390                                  PGA European Tour      N64   
16391       16391                                     Carmageddon 64      N64   
16392       16392                                    Planet Monsters      GBA   
16393       16393                                             Breach       PC   
16394       16394                                   Bust-A-Move 3000       GC   
16395       16395                                   Mega Brain Boost       DS   
16396       16396                              The Longest 5 Minutes      PSV   
16397       16397                           Mezase!! Tsuri Master DS       DS   
16398       16398  Eiyuu Densetsu: Sora no Kiseki Material Collec...      PSP   
16399       16399                     

In [137]:
df.corr()

Rank      Year  NA_Sales  EU_Sales  JP_Sales  Other_Sales  \
Rank          1.000000  0.178128 -0.400318 -0.379150 -0.269333    -0.332742   
Year          0.178128  1.000000 -0.091301  0.006151 -0.169379     0.041157   
NA_Sales     -0.400318 -0.091301  1.000000  0.768925  0.451284     0.634519   
EU_Sales     -0.379150  0.006151  0.768925  1.000000  0.436377     0.726256   
JP_Sales     -0.269333 -0.169379  0.451284  0.436377  1.000000     0.290558   
Other_Sales  -0.332742  0.041157  0.634519  0.726256  0.290558     1.000000   
Global_Sales -0.426983 -0.074637  0.941270  0.903264  0.612773     0.747964   

              Global_Sales  
Rank             -0.426983  
Year             -0.074637  
NA_Sales          0.941270  
EU_Sales          0.903264  
JP_Sales          0.612773  
Other_Sales       0.747964  
Global_Sales      1.000000

In [145]:
df_RNA = df[['NA_Sales','EU_Sales']].copy()
classes_RNA = df['Other_Sales'].copy()
display(df_RNA)
classes_RNA

NA_Sales  EU_Sales
0      1.000000  1.000000
1      0.700892  0.123363
2      0.382020  0.443832
3      0.379610  0.379394
4      0.271632  0.306340
5      0.559171  0.077877
6      0.274283  0.318057
7      0.338154  0.317023
8      0.351651  0.243280
9      0.649072  0.021709
10     0.218607  0.379049
11     0.236443  0.260855
12     0.216920  0.212957
13     0.215474  0.276706
14     0.219089  0.296003
15     0.360810  0.170227
16     0.168956  0.319435
17     0.227284  0.013784
18     0.308026  0.129221
19     0.114485  0.319090
20     0.154736  0.155755
21     0.261027  0.093384
22     0.229935  0.118539
23     0.232104  0.182977
24     0.202699  0.189180
25     0.146059  0.134390
26     0.134249  0.113025
27     0.082912  0.184700
28     0.165100  0.175396
29     0.217643  0.147484
...         ...       ...
16568  0.000000  0.000000
16569  0.000241  0.000000
16570  0.000000  0.000345
16571  0.000000  0.000000
16572  0.000000  0.000000
16573  0.000241  0.000000
16574  0.000000  0.000000
16575  0.000000  0.000000
16576  0.000000  0.000345
16577  0.000000  0.000345
16578  0.000241  0.000000
16579  0.000241  0.000000
16580  0.000000  0.000000
16581  0.000000  0.000345
16582  0.000241  0.000000
16583  0.000241  0.000000
16584  0.000241  0.000000
16585  0.000241  0.000000
16586  0.000000  0.000345
16587  0.000000  0.000000
16588  0.000241  0.000000
16589  0.000000  0.000000
16590  0.000000  0.000000
16591  0.000241  0.000000
16592  0.000241  0.000000
16593  0.000241  0.000000
16594  0.000241  0.000000
16595  0.000000  0.000000
16596  0.000000  0.000345
16597  0.000241  0.000000

[16290 rows x 2 columns]

0        0.800378
1        0.072848
2        0.313150
3        0.280038
4        0.094607
5        0.054872
6        0.274361
7        0.269631
8        0.213813
9        0.044465
10       0.260170
11       0.181646
12       0.067171
13       0.203406
14       0.169347
15       0.157994
16       0.391675
17       1.000000
18       0.052034
19       0.193945
20       0.129612
21       0.039735
22       0.043519
23       0.130558
24       0.168401
25       0.047304
26       0.077578
27       0.111637
28       0.109745
29       0.124882
           ...   
16568    0.000000
16569    0.000000
16570    0.000000
16571    0.000000
16572    0.000000
16573    0.000000
16574    0.000000
16575    0.000000
16576    0.000000
16577    0.000000
16578    0.000000
16579    0.000000
16580    0.000000
16581    0.000000
16582    0.000000
16583    0.000000
16584    0.000000
16585    0.000000
16586    0.000000
16587    0.000000
16588    0.000000
16589    0.000000
16590    0.000000
16591    0.000000
16592    0

In [146]:
x_treino, x_teste, y_treino, y_teste = train_test_split(df_RNA,classes_RNA,test_size = 0.3)

model = LinearRegression()
# 2. Use fit
model.fit(x_treino, y_treino)
# 3. Check the score
score_regressao = model.score(x_teste, y_teste)
print("Porcentagem de acerto da regressão linear: {}%".format(score_regressao*100))

Porcentagem de acerto da regressão linear: 77.2092165010196%


### Hipótese: 

#### É possível prever o número de vendas em outras regiões do mundo baseado nas vendas da América do Norte, Europa e Japão?

A correlação entre as vendas das regiões da América do Norte e Europa com as vendas de outros lugares do mundo são bem fortes, o que implica serem valores relevantes entre eles. Através da execução da Regressão Linear, observamos que a chance de acerto foi de 77% aproximadamente, confirmando a hipótese de que é possível prever o número de vendas de um jogo em outras regioes, a partir das vendas desse jogo na América do Norte e na Europa.